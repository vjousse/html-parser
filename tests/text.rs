use html_parser::{Dom, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_parse_document_with_just_text() -> Result<()> {
    let html = "hello world";
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}

#[test]
fn it_can_parse_document_with_text_and_line_breaks() -> Result<()> {
    let html = indoc!(
        r"
        hello world
        here's another line for you!
        The end
    "
    );
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}

#[test]
fn it_can_parse_document_with_multiple_text_elements() -> Result<()> {
    let html = indoc!(
        r"
        hello world
        here's another line for you!
        <div/>
        The end
    "
    );
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}

#[test]
fn it_can_parse_text_with_chevron() -> Result<()> {
    let html = indoc!(r"hello <> world");
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}

#[test]
fn it_can_parse_text_in_paragraph_with_weird_formatting() -> Result<()> {
    let html = indoc!(
        r"
        <p>
            This is a <b>para</b>gra<b>ph</b> with some<i> weird </i> formatting.
        </p>
    "
    );
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}

#[test]
fn it_can_parse_text_with_only_spaces_between_elements() -> Result<()> {
    let html = indoc!(r"This is a <b>para</b> <b>ph</b> with some<i> weird </i> formatting.");
    let dom = Dom::parse(html)?;
    assert_json_snapshot!(dom);
    Ok(())
}
