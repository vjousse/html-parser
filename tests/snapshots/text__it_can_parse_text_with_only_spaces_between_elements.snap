---
source: tests/text.rs
expression: dom
---
{
  "treeType": "documentFragment",
  "children": [
    "This is a ",
    {
      "name": "b",
      "variant": "normal",
      "children": [
        "para"
      ]
    },
    " ",
    {
      "name": "b",
      "variant": "normal",
      "children": [
        "ph"
      ]
    },
    " with some",
    {
      "name": "i",
      "variant": "normal",
      "children": [
        " weird "
      ]
    },
    " formatting."
  ]
}
